docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,872 @@
+
+##### lib
+
+This directory contains files which I cloned from the <b>agoric-sdk</b> repository and these contain functions which are used in the middleware and monitoring script.
+
+##### docker-compose.yml
+
+This is a docker-compose file to spin up the middleware and monitoring script
+
+##### Dockerfile.middleware
+
+This is a docker file to build the middleware
+
+##### Dockerfile.monitor
+
+This is a docker file to build the monitoring script
+
+##### monitoring-grafana-dashboard.json
+
+This is a Grafana template to monitor an oracle node or the whole oracle network
+
+## Technical Documentation
+
+In this section, I will go over the <b>oracle</b> directory and explain in detail each function in the files inside it.
+
+<div id='helperjs'></div>
+
+### <u>helper.js</u>
+
+This file contains helper functions which are used both by the middleware and the monitoring script.
+
+The file contains the following functions:
+
+<br>
+<div id='readJSONFile'></div>
+
+<b>readJSONFile(filename)</b>
+
+Inputs:
+* filename - This is the file name or path to the file from which to read
+
+Use: This function is used to read a JSON object from a file
+
+Returns: The JSON contents in the file
+
+What it does:
+  1. Reads the file
+  2. Parses the contents to a JSON variable
+  3. Returns the variable containing the JSON object or array
+
+<br>
+<div id='saveJSONDataToFile'></div>
+
+<b>saveJSONDataToFile(newData, filename)</b>
+
+Inputs:
+* newData - This is the JSON data to save
+* filename - This is the file name or path to the file where to save data
+
+Use: This function is used to save a JSON object/array to a file
+
+What it does:
+  1. Stringifies the data
+  2. Writes the data to the file
+
+<br>
+<div id='validUrl'></div>
+
+<b>validUrl(url)</b>
+
+Inputs:
+* url - This is the URL to check
+
+Use: This function is used to check whether a URL is valid
+
+Returns: A boolean showing whether it is a valid URL
+
+What it does:
+  1. Tryies to create a URL object
+  2. Returns whether it is a valid URL by seeing whether the URL was successfully created or not
+
+<br>
+<div id='delay'></div>
+
+<b>delay(ms)</b>
+
+Inputs:
+* ms - Milliseconds to delay
+
+Use: This function is used to create a delay
+
+Returns: A Promise with a delay of a specified number of milliseconds
+
+<br>
+<div id='readvstorage'></div>
+
+<b>readVStorage(vstorage, feed, roundData)</b>
+
+Inputs:
+* vstorage - vstorage object obtained on startup
+* feed - The feed name
+* roundData - A boolean indicating whether we are reading round data. A false means we are reading price data
+
+Use: This function is used read price or round data from vstorage
+
+Returns: CapDataString by calling vstorage.readLatest().
+
+<br>
+<div id='middleware'></div>
+
+### <u>middleware.js</u>
+
+This file contains all the functions which are used both by the middleware. 
+
+<br>
+<div id='envvarsmiddleware'></div>
+
+<b>Environment Variables</b>
```

Comment:
Very random placement to describe environment variables, put it first before explaining source code

Reply:
Monitor and Middleware use different sets of variables. This section is already before the source code explanation.